corehq/apps/users/views/mobile/users.py

```diff
@@ -1146,8 +1144,9 @@ def get(self, request, domain, *args, **kwargs):
         )
 
 
-@method_decorator([require_can_use_filtered_user_download], name='dispatch')
+@location_safe
```

Comment:
How come `FilterCommCareUserDownload` uses `@location_safe` but `FilterWebUserDownload` uses `@method_decorator([require_can_use_filtered_user_download], name='dispatch')`?

Reply:
Where do we ensure that a user in one sub-location can't download the data from a different part of the hierarchy? (e.g. a user assigned to "New York State" can't download the users in "Los Angeles"?)

Reply:
Q2 answer: [Here](https://github.com/dimagi/commcare-hq/blob/73f6cfd28e885fbe3ec3d19b1d73b708eaf65f58/corehq/apps/locations/forms.py#L649) is the part that takes the user's assigned_location_ids as part of the filters it will use to query the locations which is to be downloaded (you can also see [here](https://github.com/dimagi/commcare-hq/blob/73f6cfd28e885fbe3ec3d19b1d73b708eaf65f58/corehq/apps/locations/views.py#L1007) how it's used as "root" locations from which the drilldown will happen) 
This ensures no location is downloaded to which the user does not have access to. 

You will [see](https://github.com/dimagi/commcare-hq/blob/73f6cfd28e885fbe3ec3d19b1d73b708eaf65f58/corehq/apps/locations/forms.py#L651) that the `assigned_location_ids` is only assigned to `location_ids` when `location_id` is `None` (ie the user has not selected a location on the filters page to use as root). [Here](https://github.com/dimagi/commcare-hq/blob/f23a950da0a86301838444cc0af3c23f6846c82d/corehq/apps/reports/filters/controllers.py#L98) you can see that the locations made available for the user to choose from on the filters page is also restricted to his/her own assigned locations (the location options make use of the [LocationSelectWidget](https://github.com/dimagi/commcare-hq/blob/af9a90eb955804a2c4745eeb05b5e1ff37f4067f/corehq/apps/locations/forms.py#L589) which, if you follow the logic, winds up making the `location_search` call which filters the locations appropriately).

Reply:
Q1 answer:
FilterCommCareUserDownload (mobile workers download filter page...say that x10 fast) is made GA (hence no need for `require_can_use_filtered_user_download`), but the web users page is out of scope according to spec doc.

Reply:
Regarding Q2: I wrote a test to check a suspicion ... then took some thoughts to _#solutions-soltech-devs_ on Slack ... then followed up with PR #30981. If you're happy with that PR, then I think this one is good to go.